src/Algebra/Definitions.agda

```diff
@@ -118,5 +119,11 @@ RightCancellative _•_ = ∀ {x} y z → (y • x) ≈ (z • x) → y ≈ z
 Cancellative : Op₂ A → Set _
 Cancellative _•_ = (LeftCancellative _•_) × (RightCancellative _•_)
 
+LeftNZCancellative : A → Op₂ A → Set _
+LeftNZCancellative 0# _*_ = ∀ {x y z} → ¬ x ≈ 0# → (x * y) ≈ (x * z) → y ≈ z 
```

Comment:
I would call this `Almost(Left/Right)Cancellative`, as nothing says that the element has to be a zero. Likewise I would just call the element `e` rather than `0#` as is used elsewhere in the file.